parcels/field.py

```diff
@@ -631,47 +653,58 @@ def __getitem__(self, key):
             return _deal_with_errors(error, key, vector_type=self.vector_type)
 
 
-def _validate_dataarray(data, name):
+def _validate_uxdataarray(data: ux.UxDataArray):
     """Verifies that all the required attributes are present in the xarray.DataArray or
     uxarray.UxDataArray object.
     """
-    if isinstance(data, ux.UxDataArray):
-        # Validate dimensions
-        if not ("nz1" in data.dims or "nz" in data.dims):
-            raise ValueError(
-                f"Field {name} is missing a 'nz1' or 'nz' dimension in the field's metadata. "
-                "This attribute is required for xarray.DataArray objects."
-            )
+    # Validate dimensions
+    if not ("nz1" in data.dims or "nz" in data.dims):
+        raise ValueError(
+            "Field is missing a 'nz1' or 'nz' dimension in the field's metadata. "
+            "This attribute is required for xarray.DataArray objects."
+        )
 
-        if "time" not in data.dims:
-            raise ValueError(
-                f"Field {name} is missing a 'time' dimension in the field's metadata. "
-                "This attribute is required for xarray.DataArray objects."
-            )
+    if "time" not in data.dims:
+        raise ValueError(
+            "Field is missing a 'time' dimension in the field's metadata. "
+            "This attribute is required for xarray.DataArray objects."
+        )
 
     # Validate attributes
     required_keys = ["location", "mesh"]
     for key in required_keys:
         if key not in data.attrs.keys():
             raise ValueError(
-                f"Field {name} is missing a '{key}' attribute in the field's metadata. "
+                f"Field is missing a '{key}' attribute in the field's metadata. "
                 "This attribute is required for xarray.DataArray objects."
             )
 
-    if type(data) is ux.UxDataArray:
-        _validate_uxgrid(data.uxgrid, name)
+    _validate_uxgrid(data.uxgrid)
 
 
-def _validate_uxgrid(grid, name):
+def _validate_uxgrid(grid):
     """Verifies that all the required attributes are present in the uxarray.UxDataArray.UxGrid object."""
     if "Conventions" not in grid.attrs.keys():
         raise ValueError(
-            f"Field {name} is missing a 'Conventions' attribute in the field's metadata. "
+            "Field is missing a 'Conventions' attribute in the field's metadata. "
             "This attribute is required for uxarray.UxDataArray objects."
         )
     if grid.attrs["Conventions"] != "UGRID-1.0":
         raise ValueError(
-            f"Field {name} has a 'Conventions' attribute that is not 'UGRID-1.0'. "
+            "Field has a 'Conventions' attribute that is not 'UGRID-1.0'. "
             "This attribute is required for uxarray.UxDataArray objects."
             "See https://ugrid-conventions.github.io/ugrid-conventions/ for more information."
         )
```

Comment:
I think its nice removing `name` from here as it's only used in the error message itself. The context can be added via `add_note` (introduced in 3.11) as the message propagates up